src/main/java/io/supertokens/signingkeys/SigningKeys.java

```diff
@@ -194,6 +194,19 @@ public KeyInfo getLatestIssuedDynamicKey()
         }
     }
 
+    public long getCacheDurationInSeconds()
+            throws StorageQueryException, StorageTransactionLogicException, TenantOrAppNotFoundException,
+            UnsupportedJWTSigningAlgorithmException {
+        CoreConfig config = Config.getConfig(appIdentifier.getAsPublicTenantIdentifier(), main);
+
+        long timeLeftForNewKeyCreation = (
```

Comment:
need to double check this is correct.

Reply:
One issue here is that `getLatestIssuedDynamicKey` will return the "old" key during the overlap period, meaning this would return a negative value in that case. 
I see two main ways to fix this:
1. consider if we are in the overlap (where this calc returns a negative value) and update the cache timeout accordingly, since the jwks endpoint should already return the new key.
2. get the latest key disregarding the overlap period (I think you can get it from `getDynamicKeys` ) and just add the update interval. This should be safe if `getDynamicSigningKeyOverlapMS > 0`.
Just as a further explanation: the intention of `getDynamicSigningKeyOverlapMS` is to have a safe value for the cache timeout: anything equal to or lower than this will always see all key updates.

Reply:
Done

Reply:
@porcellus please review and confirm. @sattvikc has implemented the second one.

Reply:
@sattvikc please add tests for this.

Reply:
Tests Done

Reply:
Not exactly: you do not have to subtract `getDynamicSigningKeyOverlapMS`, but it also shouldn't cause any issues, just slightly more refreshes.

Reply:
You can leave it as is, it only takes 1 min off of the update period which is specified in hours (and it defaults to 7days). It's fine to leave as a safety buffer I guess.